lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -91,7 +91,7 @@ public function addMultiple(array $parts = [])
      */
     public function add($part)
     {
-        if ( ! empty($part) || ($part instanceof self && $part->count() > 0)) {
+        if (!empty($part) && count($part)) {
```

Comment:
I believe we still should check for `instanceof self` and only check count in this case. Otherwise, we don't know if empty collection will cast to an empty string (can be any custom class). Something like:
```php
if (empty($path)) {
    return;
}

if ($part instanceof self && !count($part)) {
    return;
}

// proceed